.codegen/api.go.tmpl

```diff
@@ -12,7 +12,92 @@ import (
 	"github.com/databricks/databricks-sdk-go/listing"
 	"github.com/databricks/databricks-sdk-go/useragent"
 )
+{{ $extendedInterface :=
+	list
+		"Clusters"
+		"CommandExecution"
+		"Dbfs"
+		"Libraries"
+		"StatementExecution"
+		"Workspace"
+}}
 {{range .Services}}
+
+type {{.PascalName}}APIInterface interface {
+	{{if (in $extendedInterface .PascalName) }}
+		{{.CamelName}}APIUtilities
+	{{end -}}
+
+	// WithImpl could be used to override low-level API implementations for unit 
+	// testing purposes with [github.com/golang/mock] or other mocking frameworks.
+	WithImpl(impl {{.PascalName}}Service) {{.PascalName}}APIInterface
+
+	// Impl returns low-level {{.PascalName}} API implementation
+	Impl() {{.PascalName}}Service
```

Comment:
Since these were useful only for mocking, we could consider removing them from the interface.

Also see https://github.com/databricks/databricks-sdk-go/tree/main/examples/mocking.